File: teachgram-frontend/src/types/post.ts
import type { User } from "./user";

export interface Post {
  id: string;
  title: string;
  description?: string;
  photoLink?: string;
  videoLink?: string;
  private: boolean;
  deleted: boolean;
  createdAt: string;
  updatedAt: string;
  likesCount: number; 
  user: User;
}

export interface CreatePostRequest {
  title: string;
  description?: string;
  photoLink?: string;
  videoLink?: string;
  private: boolean;
  userId: string; 
}

export interface PostResponse {
  id: string;
  title: string;
  description?: string;
  photoLink?: string;
  videoLink?: string;
  private: boolean;
  deleted: boolean;
  createdAt: string;
  updatedAt: string;
  likesCount: number;

  user: {
    id: string;
    userName: string;
    profileLink?: string;
  };
}

File: teachgram-backend/src/main/java/com/felipe/teachgram_backend/entity/Post.java
package com.felipe.teachgram_backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import java.time.LocalDate;

@Entity
@Table(name = "posts")
@Data
@NoArgsConstructor
@AllArgsConstructor
@SQLDelete(sql = "UPDATE posts SET deleted = true WHERE id=?")
@Where(clause = "deleted = false")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Column(nullable = false, length = 50)
    private String title;

    @Column(length = 500)
    private String description;

    private String photoLink;

    private String videoLink;

    @Column(nullable = false)
    private Boolean privatePost = false;

    private Boolean deleted = false;

    @CreationTimestamp
    private LocalDate createdAt;

    @CreationTimestamp
    private LocalDate updatedAt;

    private Integer likesCount = 0;

    private boolean isPrivate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDate.now();
        updatedAt = LocalDate.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDate.now();
    }
}